### PR TITLE
CI: yesqa

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

remove useless noqa

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -64,8 +64,7 @@ repos:
       - id: black
         name: Format code
 
-  - repo: https://github.com/PyCQA/flake8
-    rev: 3.9.2
+  - repo: https://github.com/asottile/yesqa
+    rev: v1.2.3
     hooks:
-      - id: flake8
-        name: Check PEP8
```
Comment (carmocca) [inline .pre-commit-config.yaml:71]: Can we keep flake8 in pre-commit?

I personally found it very useful to catch errors before pushing

File: tests/trainer/test_data_loading.py [hunk 2]
```diff
@@ -117,7 +117,7 @@ def on_train_start(self, *_) -> None:
 def test_replace_sampler_raises():
     trainer = Trainer()
     with pytest.raises(ValueError, match="needs to subclass `torch.utils.data.DataLoader"):
-        trainer.replace_sampler(object(), object(), mode="fit")  # noqa
+        trainer.replace_sampler(object(), object(), mode="fit")
```
Comment (carmocca) [inline tests/trainer/test_data_loading.py:120]: pycharm will complain about this line since the input type is not the expected.

the noqa should stay.

There are other false negatives, mostly for tests which are testing error raising situations

File: .pre-commit-config.yaml (hunk 1 above)
Comment (Borda) [reply .pre-commit-config.yaml:71]: but how the pre-commit bot works it fails on flake8, does it send a correction to the PR?

File: tests/trainer/test_data_loading.py (hunk 2 above)
Comment (Borda) [reply tests/trainer/test_data_loading.py:120]: then we shall add error type

File: .pre-commit-config.yaml (hunk 1 above)
Comment (carmocca) [reply .pre-commit-config.yaml:71]: It doesn't, but it isn't expected that all pre-commit hooks send corrections. For example

```
      - id: check-yaml
      - id: check-docstring-first
      - id: check-executables-have-shebangs
      - id: check-toml
      - id: check-case-conflict
      - id: check-added-large-files
```

File: tests/trainer/test_data_loading.py (hunk 2 above)
Comment (carmocca) [reply tests/trainer/test_data_loading.py:120]: Screenshot 2021-07-27 at 15 32 32

What is the error type for this one?

The problem is that different checkers report different error types, there doesn't seem to be a standardization of error codes

Comment (Borda) [reply tests/trainer/test_data_loading.py:120]: ok, can PyCharm has another mark for skipping this issue?
but it is just pycharm, or also another tool/checker is complaining?

Comment (Borda) [reply tests/trainer/test_data_loading.py:120]: and this is the only one, right?

Comment (carmocca) [reply tests/trainer/test_data_loading.py:120]: > can PyCharm has another mark for skipping this issue?

I don't know, but `noqa` works across tools.

> or also another tool/checker is complaining?

other tools may complain, for example mypy

> and this is the only one, right?

Other `noqa`s removed by this PR have the same issue

Comment (Borda) [reply tests/trainer/test_data_loading.py:120]: I see all check passing: flake8, mypy, pre-commit, ... so to me only PyCharm has an issue...

Comment (carmocca) [reply tests/trainer/test_data_loading.py:120]: we don't run `mypy` on the test directory, otherwise it would.

see this example

```python
$ cat kk.py 
def foo(x: int) -> None:
    print(x)


foo(5)

foo(object())
$ python kk.py 
5
<object object at 0x1034a44d0>
$ mypy kk.py 
kk.py:7: error: Argument 1 to "foo" has incompatible type "object"; expected "int"  [arg-type]
Found 1 error in 1 file (checked 1 source file)
$ flake8 kk.py 
```

(flake8 does not complain)

Comment (Borda) [reply tests/trainer/test_data_loading.py:120]: yes, but mypy has own command for ignore which we use elsewhere in codebase `# type: ignore`

Comment (Borda) [reply tests/trainer/test_data_loading.py:120]: btw, if there were only about 5 issues in the test folder (about 5 were removed) why we did not run mypy on all codebase?

**Review by ethanwharris — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]

File: tests/trainer/test_data_loading.py (hunk 2 above)
Comment (carmocca) [reply tests/trainer/test_data_loading.py:120]: We don't run mypy for the tests folder

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -76,3 +76,9 @@ repos:
     hooks:
       - id: flake8
         name: Check PEP8
+
+  - repo: https://github.com/asottile/yesqa
+    rev: v1.2.3
+    hooks:
+      - id: yesqa
+        name: Unused noqa
```
Comment (carmocca) [inline .pre-commit-config.yaml:84]: Since this modified code and can impcat pep8, this hook should run before isort, black and flake8

**Review by carmocca — APPROVED** [review]
